### PR TITLE
#53908: AddListLeadComponent

### PR DESCRIPTION
**Pull request by josepedrazas**
- Create RequestLead
- Create Model Lead
- Add Support Async

### CODE REVIEW

**Comment by kleyla** [comment]
well done @josepedrazas  ! I have added some features in order to look like the requirements and about the test in which you ask for help, I try to add test but it throws an error that I couldn't fix so I deleted it and finally I have added a description for table for fix the bug which appears in sonar.

**Review by kleyla — APPROVED** [review]

File: src/components/leads/Leads.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<table class="table table-bordered" aria-describedby="table of leads">
+  <caption>
+    Table of Leads
```
Comment (Carlos-YD) [inline src/components/leads/Leads.html:3]: Please move the hardcoded title of the table and the headers like you did previously with the hardcoded **CRM** text.

**Review by Carlos-YD — CHANGES_REQUESTED** [review]

**Comment by Carlos-YD** [comment]
> SonarCloud Quality Gate failed.    Quality Gate failed
> 
> [Bug](https://sonarcloud.io/project/issues?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=BUG)
> [Vulnerability](https://sonarcloud.io/project/issues?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=VULNERABILITY)
> [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=SECURITY_HOTSPOT)
> [Code Smell](https://sonarcloud.io/project/issues?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=CODE_SMELL) [0 Code Smells](https://sonarcloud.io/project/issues?id=FJala-Tr2020_front-end-e2&pullRequest=3&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=FJala-Tr2020_front-end-e2&pullRequest=3&metric=new_coverage&view=list) [72.7% Coverage](https://sonarcloud.io/component_measures?id=FJala-Tr2020_front-end-e2&pullRequest=3&metric=new_coverage&view=list)
> [](https://sonarcloud.io/component_measures?id=FJala-Tr2020_front-end-e2&pullRequest=3&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=FJala-Tr2020_front-end-e2&pullRequest=3&metric=new_duplicated_lines_density&view=list)

The SonarCloud quality gate failed, you reached 72% of code coverage and you need at least 80% to allow merging the code.

File: src/components/leads/Leads.html (hunk 1 above)
Comment (kleyla) [reply src/components/leads/Leads.html]: Changes done.. thank you

Comment (Carlos-YD) [reply src/components/leads/Leads.html]: We will update this later, it is OK to extract it to the class but it does not make sense to read it from an env. var.

**Review by Carlos-YD — APPROVED** [review]

**Comment by Carlos-YD** [comment]
Just resolve the conflicts and we will merge the code.

**Comment by kleyla** [comment]
Thank you.. I have solved the conflicts..

Can you create a file '.env' on the root of the frontend with the environment variables that are

```
BASE_URL=https://cainco-backend-e2.herokuapp.com/
DESCRIPTION_TABLE_LEADS='Table of Leads'
```

**Pull request merged**